server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -197,12 +197,19 @@ public String getSetting(String index, String setting) {
     public void writeTo(StreamOutput out) throws IOException {
         out.writeStringArray(indices);
         out.writeVInt(mappings.size());
-        for (ObjectObjectCursor<String, ImmutableOpenMap<String, MappingMetaData>> indexEntry : mappings) {
+        for (ObjectObjectCursor<String, MappingMetaData> indexEntry : mappings) {
             out.writeString(indexEntry.key);
-            out.writeVInt(indexEntry.value.size());
-            for (ObjectObjectCursor<String, MappingMetaData> mappingEntry : indexEntry.value) {
-                out.writeString(mappingEntry.key);
-                mappingEntry.value.writeTo(out);
+            if (out.getVersion().before(Version.V_8_0_0)) {
+                out.writeVInt(indexEntry.value == MappingMetaData.EMPTY_MAPPINGS ? 0 : 1);
+                if (indexEntry.value != null) {
```

Comment:
Should this be `indexEntry.value != MappingMetaData.EMPTY_MAPPINGS`?

Reply:
Missed this one... will fix, thanks!